### PR TITLE
Add plugin-list-all command

### PR DESCRIPTION
**Pull request by tuvistavie**
`asdf plugin-list-all` will show a list of all the available commands in the plugin repository.

### CODE REVIEW

**Comment by vic** [comment]
Refs #211

File: lib/utils.sh [hunk 1]
```diff
@@ -261,20 +261,31 @@ asdf_repository_url() {
   echo "https://github.com/asdf-vm/asdf-plugins.git"
 }
 
+repository_needs_update() {
+  local update_file_dir="$(asdf_dir)/tmp"
+  local update_file_name="repo-updated"
+  # `find` outputs filename if it has not been modified in the last day
+  local find_result=$(find $update_file_dir -name "$update_file_name" -type f -mtime +1 -print)
+  [ -n "$find_result" ]
```
[the remark this answers is not in the document]
Comment (vic) [reply lib/utils.sh]: Looks like we dont :)

**Comment by tuvistavie** [comment]
Thank you!

File: lib/utils.sh [hunk 2]
```diff
@@ -261,20 +261,31 @@ asdf_repository_url() {
...
   local repository_path
 
   repository_url=$(asdf_repository_url)
   repository_path=$(asdf_dir)/repository
 
-  if [ -d "$repository_path" ]; then
-    echo "updating plugin repository..."
-    (cd "$repository_path" && git fetch && git reset --hard origin/master)
-  else
+  if [ ! -d "$repository_path" ]; then
```
Comment (Stratus3D) [inline lib/utils.sh]: Why this change? The previous if condition was better because it didn't negate the directory check.

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,7 @@
 installs
-plugins
```
[the remark this answers is not in the document]
Comment (vic) [reply .gitignore]: Having `installs` and `plugins` in `.gitignore` allows us to use a local development branch of asdf directly (for manual testing for example), add plugins into it and install some of their versions.

[the remark this answers is not in the document]
Comment (tuvistavie) [reply .gitignore]: @Stratus `test/fixtures/dummy_plugins_repo/plugins/bar` was being ignored.

File: lib/utils.sh (hunk 2 above)
Comment (tuvistavie) [reply lib/utils.sh]: I wanted to avoid fetching the repository every time.
With the new condition, repository will only be fetched once a day. 
I do not think to be neglecting the directory check here, what do you mean?

Comment (Stratus3D) [reply lib/utils.sh]: My bad, I missed the new `elif` below.
